src/utils/components/InstanceTypeModal/ConnectedInstanceTypeModal.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { FC } from 'react';
+
+import useInstanceTypesAndPreferences from '@catalog/CreateFromInstanceTypes/state/hooks/useInstanceTypesAndPreferences';
+
+import ErrorAlert from '../ErrorAlert/ErrorAlert';
+import Loading from '../Loading/Loading';
+
+import InstanceTypeModal, { InstanceTypeModalProps } from './InstanceTypeModal';
+
+const ConnectedInstanceTypeModal: FC<
+  Pick<InstanceTypeModalProps, 'instanceTypeVM' | 'isOpen' | 'onClose' | 'onSubmit'>
+> = ({ instanceTypeVM, ...otherProps }) => {
+  const { allInstanceTypes, loaded, loadError } = useInstanceTypesAndPreferences();
+  const instanceType = allInstanceTypes.find(
+    (it) => it.metadata.name === instanceTypeVM?.spec?.instancetype?.name,
+  );
+  return (
+    <>
+      {!loaded && <Loading />}
+      {loaded && loadError && <ErrorAlert error={loadError} />}
+      {loaded && !loadError && (
+        <InstanceTypeModal
+          {...otherProps}
```

Comment:
as a rule of thumb , using {...otherProps} , should be avoided.
For example, lets say other props contiant 3 props.
if InstanceTypeModal , has useMemo, useCallback, that listen to only 1 prop from otherProps, it will recalc even if the 2 other props are getting new values, as {...otherProps} will always create a new object, new reference so useMemo and useCallback will fail compare and recalc. better to explicit pass props.

Reply:
I will check it but we should be OK here. The component receives individual props which have the same references.

Reply:
Works for me.